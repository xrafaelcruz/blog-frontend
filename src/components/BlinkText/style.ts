import styled, { keyframes } from 'styled-components'

export const Wrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 0 0 40px;
  width: 100%;

  @media (max-width: 920px) {
    display: none;
  }
`

const blinkCursor = keyframes`
  from {
    border-color: rgba(255, 255, 255, 0.75);
  }

  to {
    border-color: transparent;
  }
`

const typing = keyframes`
  from {
    width: 0;
  }

  to {
    width: 55rem;
  }
`

export const Text = styled.h1`
  && {
    align-self: center;
    animation: ${blinkCursor} 500ms steps(40) infinite normal,
      ${typing} 2s steps(40) 1s normal both;
    border-right: 2px solid rgba(255, 255, 255, 0.75);
    display: inline-block;
    font-family: ${(props) => props.theme.font.blink};
    justify-content: center;
    margin: 1rem auto;
    overflow: hidden;
    white-space: nowrap;
    text-align: center;

    @media (max-width: 920px) {
      animation: none;
    }
  }
`
