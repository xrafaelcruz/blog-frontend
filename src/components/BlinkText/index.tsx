import * as S from './style'

export type Props = {
  children: string
}

const BlinkText = ({ children }: Props) => (
  <S.Wrapper>
    <S.Text>{children}</S.Text>
  </S.Wrapper>
)

export default BlinkText
