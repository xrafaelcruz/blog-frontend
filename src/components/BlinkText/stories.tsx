import React from 'react'
import { Story, Meta } from '@storybook/react'
import BlinkText, { Props } from '.'

export default {
  title: 'BlinkText',
  component: BlinkText,
  args: {
    children: 'Bem vindo ao meu blog front-end!'
  }
} as Meta

export const Default: Story<Props> = (args) => <BlinkText {...args} />
