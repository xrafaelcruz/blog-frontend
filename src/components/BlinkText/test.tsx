import 'jest-styled-components'
import { screen } from '@testing-library/react'
import { renderWithTheme } from '../../utils/tests/helpers'
import BlinkText from '.'

describe('<BlinkText />', () => {
  const text = 'Bem vindo ao meu blog front-end!'
  const heading = { level: 1, name: text }
  const Component = <BlinkText>{text}</BlinkText>

  it('should render the text of BlinkText', () => {
    renderWithTheme(Component)
    expect(screen.getByRole('heading', heading))
  })

  it('should not render the BlinkText on mobile', () => {
    renderWithTheme(Component)

    const element = screen.getByRole('heading', heading).parentElement

    expect(element).toHaveStyleRule('display', 'none', {
      media: '(max-width: 920px)'
    })

    expect(element).toMatchSnapshot()
  })
})
