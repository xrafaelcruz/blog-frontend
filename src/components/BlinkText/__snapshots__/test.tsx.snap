// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlinkText /> should not render the BlinkText on mobile 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0 40px;
  width: 100%;
}

.c1.c1 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-animation: iCyueR 1500ms steps(40) infinite normal,kxrAqI 2s steps(40) 1s normal both;
  animation: iCyueR 1500ms steps(40) infinite normal,kxrAqI 2s steps(40) 1s normal both;
  border-right: 2px solid rgba(255,255,255,0.75);
  display: inline-block;
  font-family: Special Elite;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 1rem auto;
  overflow: hidden;
  white-space: nowrap;
  text-align: center;
}

@media (max-width:920px) {

}

@media (max-width:920px) {

}

@media (max-width:920px) {
  .c0 {
    display: none;
  }
}

@media (max-width:920px) {
  .c1.c1 {
    -webkit-animation: none;
    animation: none;
  }
}

<div
  class="c0"
>
  <h1
    class="c1"
  >
    Bem vindo ao meu blog front-end!
  </h1>
</div>
`;
